go/chat/boxer_test.go

```diff
@@ -205,6 +205,173 @@ func TestChatMessageUnbox(t *testing.T) {
 	})
 }
 
+func TestChatMessageUnboxWithPairwiseMacs(t *testing.T) {
+	key := cryptKey(t)
+	text := "hi"
+	tc, boxer := setupChatTest(t, "unbox")
+	defer tc.Cleanup()
+
+	u, err := kbtest.CreateAndSignupFakeUser("unbox", tc.G)
+	if err != nil {
+		t.Fatal(err)
+	}
+	msg := textMsgWithSender(t, text, gregor1.UID(u.User.GetUID().ToBytes()), chat1.MessageBoxedVersion_V2)
+	outboxID := chat1.OutboxID{0xdc, 0x74, 0x6, 0x5d, 0xf9, 0x5f, 0x1c, 0x48}
+	msg.ClientHeader.OutboxID = &outboxID
+	// Pairwise MACs rely on the sender's DeviceID in the header.
+	deviceID := make([]byte, libkb.DeviceIDLen)
+	err = tc.G.ActiveDevice.DeviceID().ToBytes(deviceID)
+	require.NoError(t, err)
+	msg.ClientHeader.SenderDevice = gregor1.DeviceID(deviceID)
+
+	signKP := getSigningKeyPairForTest(t, tc, u)
+
+	encryptionKeypair, err := tc.G.ActiveDevice.NaclEncryptionKey()
+	require.NoError(t, err)
+	pairwiseMACRecipients := []keybase1.KID{encryptionKeypair.GetKID()}
+
+	boxed, err := boxer.box(context.TODO(), msg, key, nil, signKP, chat1.MessageBoxedVersion_V2, pairwiseMACRecipients)
+	require.NoError(t, err)
+	boxed = remarshalBoxed(t, *boxed)
+
+	if boxed.ClientHeader.OutboxID == msg.ClientHeader.OutboxID {
+		t.Fatalf("defective test: %+v   ==   %+v", boxed.ClientHeader.OutboxID, msg.ClientHeader.OutboxID)
+	}
+
+	// need to give it a server header...
+	boxed.ServerHeader = &chat1.MessageServerHeader{
+		Ctime: gregor1.ToTime(time.Now()),
+	}
+
+	unboxed, err := boxer.unbox(context.TODO(), *boxed, chat1.ConversationMembersType_KBFS, key, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	body := unboxed.MessageBody
+	if typ, _ := body.MessageType(); typ != chat1.MessageType_TEXT {
```

Comment:
`require.Equal`?

Reply:
throughout these tests would be ideal since it's easier to read the tests without the cruft but won't block merge on it